distributed/deploy/local_env.py

```diff
@@ -0,0 +1,279 @@
+import asyncio
+import logging
+import weakref
+from typing import List, Union
+
+import dask
+import dask.config
+
+from ..core import Status
+from ..scheduler import Scheduler as _Scheduler
+from ..utils import cli_keywords
+from ..worker import Worker as _Worker
+from .spec import ProcessInterface, SpecCluster
+
+logger = logging.getLogger(__name__)
+
+
+class Process(ProcessInterface):
+    """A superclass for SSH Workers and Nannies
+
+    See Also
+    --------
+    Worker
+    Scheduler
+    """
+
+    def __init__(self, **kwargs):
+        self.proc = None
+        super().__init__(**kwargs)
+
+    async def start(self):
+        weakref.finalize(self, self.proc.kill)
+        await super().start()
+
+    async def close(self):
+        self.proc.kill()
+        await super().close()
+
+    def __repr__(self):
+        return "<SSH %s: status=%s>" % (type(self).__name__, self.status)
+
+
+class Worker(Process):
+    """A Remote Dask Worker run by a specified Python executable
+
+    Parameters
+    ----------
+    scheduler: str
+        The address of the scheduler
+    python_exe: str
+        Full path to Python executable to run this worker
```

Comment:
Small nit. The name `python_exe` feels a bit windowsy to me.

The `SSHCluster` calls it `remote_python` which doesn't make sense here. But I wonder if `python_bin` or just `python` would make sense. 

I note that `sys` calls it `sys.executable` to maybe the full `python_executable` would also feel more consistent.

Reply:
I followed `sys.executable` for the naming of this - although I couldn't find a more generic name for it. It's either a 'binary' or an 'executable' depending on your OS of choice. I think I like `python_executable` best of the other options you suggest, so maybe I'll go with that...